src/docfx/lib/markdown/MarkdigUtility.cs

```diff
@@ -153,6 +154,56 @@ public static string ToPlainText(this MarkdownObject containerBlock)
             return writer.ToString();
         }
 
+        public static string ToHtml(this MarkdownObject markdownObject)
```

Comment:
This way of check visibility is very likely to have visible perf impact if applied to every node (Equivalent to render markdown AST twice).